packages/slate-commons/src/types/DecorateFactory.ts

```diff
@@ -1,9 +1,5 @@
 import type { Editor } from 'slate';
 
 import type { Decorate } from './Decorate';
-import type { Extension } from './Extension';
 
-export type DecorateFactory = <E extends Editor, X extends Extension>(
-    editor: E,
-    extension: X,
-) => Decorate;
```

Comment:
We never relied on the `extension` being passed to these. So I've dropped the parameter to simplify the code and reuse the interface in `useDecorationFactory()`, which wouldn't be possible otherwise.